src/main/scala/gov/nih/nlm/ncbi/finagle/consul/catalog/ConsulCatalogResolver.scala

```diff
@@ -52,12 +52,20 @@ class ConsulCatalogResolver extends Resolver {
   }
 
   private def jsonToAddresses(json: JValue): Set[InetSocketAddress] = {
-    json
-      .extract[Set[HealthJson]]
-      .map { ex => new InetSocketAddress(Option(ex.Service.Address).filterNot(_.isEmpty).getOrElse(ex.Node.Address), ex.Service.Port) }
+    val result = Try {
+      json
+        .extract[Set[HealthJson]]
+        .map { ex => new InetSocketAddress(Option(ex.Service.Address).filterNot(_.isEmpty).getOrElse(ex.Node.Address), ex.Service.Port) }
+    }
+    result match {
+      case Return(addresses) => addresses
+      case Throw(t) =>
```

Comment:
This changes the behaviour instead of throwing exception we now return empty Set, is this change safe?

Reply:
The motivation for the change being that I suspect that we have a parse error that silently shuts down the resolver. After this change, if it's present it won't be silent, and the behavior is no worse than what we had before.

It can be argued that it's better to try to return the previous value, if you think that's the case, I'll try to adapt.

Reply:
The motivation for the change being that I suspect that we have a parse error that silently shuts down the resolver. After this change, if it's present it won't be silent, and the behavior is no worse than what we had before.

It can be argued that it's better to try to return the previous value, if you think that's the case, I'll try to adapt.

Reply:
Changed the code so that when the parsing of the JSON fails there's no update to the current set of addresses.